File: src/Scripting/PHP/patterns/model_view_presenter.php
<?php declare(strict_types=1);
$check = static fn(bool $ok) => $ok ?: throw new RuntimeException('mvp');
$model = static fn(): array => ['name'=>'Ada']; $rendered='';
$view = static function(string $text) use (&$rendered): void { $rendered=$text; };
$presenter = static fn() => $view('Hello '.$model()['name']);
$presenter(); $check($rendered === 'Hello Ada');

File: src/Scripting/Ruby/patterns/model_view_presenter.rb
# frozen_string_literal: true

module RubyPatterns
  module ModelViewPresenter
    module_function

    def run
      model = { name: 'Ada' }
      view = {}
      presenter = -> { view[:text] = model[:name].upcase }
      presenter.call
      raise 'Model-View-Presenter failed' unless view[:text] == 'ADA'
      true
    end
  end
end

RubyPatterns::ModelViewPresenter.run if $PROGRAM_NAME == __FILE__
